Sources/SwiftSyntax/SwiftSyntaxAssert.swift

```diff
@@ -0,0 +1,35 @@
+//===---------------------- SwiftSyntaxAssert.swift -----------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+fileprivate let forceEnableAssertions: Bool = {
+	ProcessInfo.processInfo.environment["SWIFT_SYNTAX_FORCE_ENABLE_ASSERTIONS"] != nil
+}()
```

Comment:
You could make this ever-so-slightly cheaper by making this a simple public Bool variable and doing the environment check once at the top of your main function.  That would avoid the additional lazy initialization check.  You should benchmark and see if such a change actually makes a measurable difference, of course.